src/Manager/Indices.php

```diff
@@ -158,10 +170,19 @@ private function getStopwordFilterKey( $filtersList ) {
 	 */
 	public function generateIndexByIndexable( $indexable ) {
 		$indexName = $indexable->get_index_name();
+		// if document pipeline does not exist create it
+		if ( ! $this->pipelinesManager->documentPipelineExists( $indexName ) && $this->documentsFeature->is_active()) {
```

Comment:
```suggestion
		if ( ! $this->pipelinesManager->documentPipelineExists( $indexName ) && $this->documentsFeature->is_active() ) {
```

Could we have a small optimization by swapping the conditions @jdtobonf?

```php
		if ( $this->documentsFeature->is_active() && ! $this->pipelinesManager->documentPipelineExists( $indexName ) ) {
```

I am suggesting that because I see `documentPipelineExists` is doing some kind of request.

Reply:
Updated